enable.ps1

```diff
@@ -148,151 +69,379 @@ function Resolve-ZenyaErrorMessage {
-    ExportData = [PSCustomObject]@{
-        id       = $aRef.id
-        userName = $aRef.userName
+                # Add the updated properties to account body
+                foreach ($newProperty in $newProperties) {
+                    # Add department as extension property to account body
+                    if ($newProperty.Name -eq "Department") {
+                        $accountBody.operations += @{ 
+                            op    = "replace"
+                            path  = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:Department"
+                            value = $newProperty.Value
+                        }
+                    }
+                    # Add manager as extension property to account body
+                    elseif ($newProperty.Name -eq "Manager") {
+                        $accountBody.operations += @{ 
+                            op    = "replace"
+                            path  = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:Manager"
+                            value = $newProperty.Value
+                        }
+                    }
+                    # Transform and add emails to account body
+                    elseif ($newProperty.Name -eq "Emails") {
+                        foreach ($email in $newProperty.Value) {
+                            if ($email.StartsWith("work:")) {
+                                $accountBody.operations += @{      
+                                    op    = "replace"
+                                    path  = "emails[type eq `"work`"].value"
+                                    value = $email -replace "work:", ""
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        $accountBody.operations += @{
+                            op    = "replace"
+                            path  = $newProperty.name
+                            value = $newProperty.value
+                        }
+                    }
+                }
+
+                $body = ($accountBody | ConvertTo-Json -Depth 10)
+                $splatWebRequest = @{
+                    Uri             = "$baseUrl/scim/users/$($currentAccount.Id)"
+                    Headers         = $headers
+                    Method          = 'PATCH'
+                    Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+                    ContentType     = "application/json;charset=utf-8"
+                    UseBasicParsing = $true
+                }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    Write-Verbose "Updating account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Body: $($updateAccount | ConvertTo-Json -Depth 10)"
+    
+                    $updatedAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
```

Comment:
here I also don't see the enable attribute, is this done in the mapping? If so, we would like to know how to configure this.

Reply:
Done in mapping of field "Active", example mapping will be added to the readme.